docs/DeveloperGuide.md

```diff
@@ -2,32 +2,143 @@
+For successful searches, the program will output the relevant books along with the book details: ISBN, Title, Author, Topic. For unsuccessful searches, the program will output a string informing the user that there is no match with the input topic from the inventory.
+
+
+### [Proposed]  Borrow book
+Description: Borrow a book from the library  
+Format: borrow -title TITLE
+Example:
+- borrow -title Python Programming  
+For successful borrowing, the program will output a string showing that the action is successful and, at the same time, mark the book as borrowed in the system. For unsuccessful borrow requests, the program will either output that there is no such book in the inventory or a message showing that the book is already on loan at the time of the borrow request.
+
+
+### [Proposed]  Check book availability
+Description: Check if a book is available for borrowing  
+Format: avail -title TITLE
+Example:
+- check -title Python Programming  
+The program will indicate whether the book is available for borrowing. The program will also handle the case that there is no such book in the inventory.
+
+### [Proposed]  Renew borrow period of book
+Description: Renew borrowing of books for a fixed duration  
+Format: renew -title TITLE
+Example:
+- renew -title C++Primer  
+For successful renewal of books, the program will output a string showing that the action is successful and also change the due period of borrow in the system. The program will handle error cases such as incorrect titles provided or books not available for renewal.
+
+
+### [Proposed]  Check borrowing status
+Description: Check status of borrowed book  
+Format: status -title TITLE
+Example:
+- status -title C++Primer  
+The program will output the details of the relevant book being borrowed and also show the due date of the loan. The program will handle cases where there is no such book in the borrow history.
+
+
+### [Proposed]  Return book
+Description: Return a book to the library  
+Format: return -title TITLE
+Example:
+- return -title C++Primer
+For successful returns of books, the program will output a string showing that the action is successful and mark the book as available for borrowing by other users in the system. The program will handle cases of incorrect title input and unsuccessful returns.
+
+
+### [Proposed] Add book
+![](images/Inventory%20class%20diagram.png)
+Description: Add new books into the system  
+Format: librarian -title TITLE -topic TOPIC -author AUTHOR -isbn ISBN -action add
+Example:
+- librarian -title C++Primer -topic Programming -author James -isbn 12345 -action add  
+This feature is only applicable for admin. For successful addition of new book, the program will output a message to inform the librarian. The new book will also be added into the inventory of the system so that users can now borrow the book from the library. Error inputs and missing inputs will be handled accordingly.
+ 
+
+### [Proposed] Delete book
```

Comment:
perhaps you can create a sequence diagram for each of these command steps to share how the data is changed when such commands are used